src/instructlab/model/backends/backends.py

```diff
@@ -29,3 +32,58 @@ def run(self):
     @abc.abstractmethod
     def shutdown(self):
         """Shutdown serving backend"""
+
+
+def is_model_gguf(model_path: pathlib.Path) -> bool:
+    """
+    Check if the file is a GGUF file.
+    Args:
+        file_path (Path): The path to the file.
+    Returns:
+        bool: True if the file is a GGUF file, False otherwise.
+    """
+    # Third Party
+    import gguf
+
+    try:
+        gguf.GGUFReader(model_path, "r")
```

Comment:
That approach is very slow. It takes almost 2 seconds on my laptop. I guess it loads the entire file into memory. IMO it's sufficient to check that the file is present and has the right magic header:

```
>>> from gguf.constants import GGUF_MAGIC
>>> import struct
>>> with open("models/merlinite-7b-lab-Q4_K_M.gguf", "rb") as f:
...     buf = f.read(4)
>>> header = struct.unpack("<I", buf)
>>> header[0] == GGUF_MAGIC
```